src/platform/i386/kernel.c

```diff
@@ -68,14 +72,19 @@ kern_memory_setup(struct multiboot *mb, u32_t mboot_magic)
 	for (i = 0 ; i < mb->mods_count ; i++) {
 		struct multiboot_mod_list *mod = &mods[i];
 
-		printk("\t- %d: [%08x, %08x)", i, mod->mod_start, mod->mod_end);
+		char *cmdline = (char *)mod->cmdline;
+		int cmdline_len = strnlen((const char*)cmdline, CMDLINE_MAX_LEN);
+		assert(cmdline_len >= CMDLINE_REQ_LEN);
+		int addr_offset = cmdline_len - CMDLINE_REQ_LEN;
+
+		printk("\t- %d: [%08x, %08x) : %s", i, mod->mod_start, mod->mod_end, mod->cmdline);
 		/* These values have to be higher-half addresses */
 		glb_memlayout.mod_start = chal_pa2va((paddr_t)mod->mod_start);
 		glb_memlayout.mod_end   = chal_pa2va((paddr_t)mod->mod_end);
 
-		glb_memlayout.bootc_vaddr = (void*)hextol((char*)mod->cmdline);
-		assert(((char*)mod->cmdline)[8] == '-');
-		glb_memlayout.bootc_entry = (void*)hextol(&(((char*)mod->cmdline)[9]));
+		glb_memlayout.bootc_vaddr = (void*)hextol((char *)(cmdline + addr_offset));
```

Comment:
I've not replaced the previous logic, I've only modified to also accommodate bare-metal command line. 
When we run from Qemu, mod->cmdline will only be XXXXXXXX-YYYYYYYY but for Grub, we give the absolute path to the module and we have them in /boot/. So, I'm only stripping down the first "/boot/" if it exists. 

This logic works both on Qemu and Bare-metal as I've been testing every change on both.